core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -598,47 +561,72 @@ class GroupMetadataManager(val brokerId: Int,
     val startMs = time.milliseconds()
     var offsetsRemoved = 0
 
-    groupMetadataCache.foreach { case (groupId, group) =>
+    val result = groupMetadataCache.flatMap { case (groupId, group) =>
       group synchronized {
         if (!group.is(Dead)) {
           val offsetsPartition = partitionFor(groupId)
           val magicValueAndTimestampOpt = getMessageFormatVersionAndTimestamp(offsetsPartition)
           magicValueAndTimestampOpt match {
             case Some((magicValue, timestamp)) =>
               // delete the expired offsets from the table and generate tombstone messages to remove them from the log
-              val tombstones = group.removeExpiredOffsets(startMs).map { case (topicPartition, offsetAndMetadata) =>
+              var tombstones = group.removeExpiredOffsets(startMs).map { case (topicPartition, offsetAndMetadata) =>
```

Comment:
Do you really want this to be a var? A mutable collection is being assigned, so a var would only make sense if you want to avoid mutating the mutable collection. Is that the case here?